multicore_runtime/dispatcher.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""A WSGI app that, once configured, dispatches requests to user apps."""
+
+import logging
+import re
+
+from werkzeug.wrappers import Request
```

Comment:
One general comment (for here and elsewhere in the PR) is that we usually recommend importing *modules* and not individual classes or functions from the modules.

This makes it a lot more obvious at the spots where the classes are used to indicate where they are coming from.

Here's the rationale in the style guide:
https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Imports#Imports

My recommendation would be then:

```python
  import werkzeug.wrappers
  ...
  request = werkzeug.wrappers.Request(wsgi_env)
  ...
  response = werkzeug.wrappers.Response(
      '<h1>500 Internal Server Error</h1>\n',
      httplib.INTERNAL_SERVER_ERROR)
```

Of course, there are exceptions to the rule. Can you justify why you think you should import Request, Response directly here instead of using the module-qualified names?

The same is true elsewhere, for instance in `wsgi.py` there are many imports of individual functions and classes from other local modules, is there a strong reason for importing them all in the namespace? (I'll mark that one in particular too.)